src/js/module/K2previousissues/PreviousIssuesButton.js

```diff
@@ -0,0 +1,227 @@
+                                targetQuarter += 4;
+                            }
+                        } else if (buttonConfig.yearsAgo) {
+                            targetYear -= buttonConfig.yearsAgo;
+                        }
+
+                        previousPeriodFormatted = `Q${targetQuarter} ${targetYear}`;
+                    } else {
+                        console.debug('Cannot go back by months for a quarterly issue');
+                        this.showErrorMessage('Cannot use monthly navigation on a quarterly issue');
+                        return;
+                    }
+                } else {
+                    // Handle monthly format
+                    const currentMonth = moment(periodText, 'MMM YYYY').toDate();
+
+                    if (buttonConfig.type === 'monthly' || buttonConfig.type === 'all') {
+                        const previousMonth = new Date(currentMonth);
+                        if (buttonConfig.monthsAgo) {
+                            previousMonth.setMonth(previousMonth.getMonth() - buttonConfig.monthsAgo);
+                        } else if (buttonConfig.yearsAgo) {
+                            previousMonth.setFullYear(previousMonth.getFullYear() - buttonConfig.yearsAgo);
+                        }
+                        previousPeriodFormatted = moment(previousMonth).format('MMM YYYY');
+                    } else {
+                        console.debug('Cannot go back by quarters for a monthly issue');
+                        this.showErrorMessage('Cannot use quarterly navigation on a monthly issue');
+                        return;
+                    }
+                }
+
+                return API.getPreviousInstancesOfIssue(filteredSearchParts)
+                    .then((previousIssuesResponse) => {
+                        if (!previousIssuesResponse || _.isEmpty(previousIssuesResponse)) {
+                            console.debug('No previous issues found');
+                            this.showErrorMessage(`No previous issues found for ${previousPeriodFormatted}`);
+                            return;
+                        }
+
+                        // Convert object response to array of issues
+                        const issues = _.values(previousIssuesResponse);
+
+                        const matchingIssue = _.find(issues, issue => _.has(issue, 'milestone')
+                            && _.has(issue.milestone, 'title')
+                            && issue.milestone.title.includes(previousPeriodFormatted));
+
+                        if (!matchingIssue) {
+                            console.debug('No matching issue found for:', previousPeriodFormatted);
+                            this.showErrorMessage(`No matching issue found for ${previousPeriodFormatted}`);
+                            return;
+                        }
+
+                        // Use matching issue if found, otherwise use the first issue
```

Comment:
I don't see the logic for this part of the comment, did I miss it? Or was it old logic?

```suggestion
                        // Use matching issue if found
```